src/components/QuranReader/TafsirView/TafsirBody.tsx

```diff
@@ -129,6 +129,40 @@ const TafsirBody = ({
     fetcher,
   );
 
+  const onSelectLanguage = useCallback(
+    (newSelectedLanguage: string) => {
+      logValueChange('tafsir_locale', selectedLanguage, newSelectedLanguage);
+      setSelectedLanguage(newSelectedLanguage);
+      const tafsirsInNewSelectedLanguage = tafsirSelectionList?.tafsirs?.filter(
+        (tafsir) => tafsir.languageName === newSelectedLanguage,
+      );
+      const slug = tafsirsInNewSelectedLanguage[0]?.slug;
+      if (slug) {
+        setSelectedTafsirIdOrSlug(slug);
+        onSettingsChange(
+          'selectedTafsirs',
+          [slug],
+          setSelectedTafsirs({
+            tafsirs: [slug],
+            locale: lang,
+          }),
+          setSelectedTafsirs({
+            tafsirs: tafsirsState.selectedTafsirs,
+            locale: lang,
+          }),
+          PreferenceGroup.TAFSIRS,
+        );
```

Comment:
Salam alykum Arif,

JAK for your contribution, much appreciated!

Just a comment here. When you change the language, you set the tafsir to the first one in the new language. However, the URL doesn't change to reflect the new tafsir.

You can refactor this function to directly use `onTafsirSelected` and pass the `id` and `slug` of the new tafsir.